src/libraries/Microsoft.Extensions.Hosting.Abstractions/src/IApplicationLifetime.cs

```diff
@@ -14,30 +14,6 @@ namespace Microsoft.Extensions.Hosting
     /// </para>
     /// </summary>
     [Obsolete("IApplicationLifetime has been deprecated. Use Microsoft.Extensions.Hosting.IHostApplicationLifetime instead.")]
-    public interface IApplicationLifetime
-    {
-        /// <summary>
-        /// Triggered when the application host has fully started and is about to wait
-        /// for a graceful shutdown.
-        /// </summary>
-        CancellationToken ApplicationStarted { get; }
-
-        /// <summary>
-        /// Triggered when the application host is performing a graceful shutdown.
-        /// Requests may still be in flight. Shutdown will block until this event completes.
-        /// </summary>
-        CancellationToken ApplicationStopping { get; }
-
-        /// <summary>
-        /// Triggered when the application host is performing a graceful shutdown.
-        /// All requests should be complete at this point. Shutdown will block
-        /// until this event completes.
-        /// </summary>
-        CancellationToken ApplicationStopped { get; }
-
-        /// <summary>
-        /// Requests termination of the current application.
-        /// </summary>
-        void StopApplication();
-    }
+    public interface IApplicationLifetime : IHostApplicationLifetime
+    { }
```

Comment:
Isn't deleting all these members a breaking change?

Reply:
They aren't deleted, they are "pushed down".  What happened in the past was the interface was renamed.  At first I added them back but then realized I couldn't use them in other API.  So I instead added them back as derived from their new name.  We use this trick in classes - pushing a member to a base class is a compatible change (both at compile and runtime).  I tested and this works for implicitly implemented interfaces, but it looks like the runtime isn't happy with it if the interface is explicitly implemented.  Probably that could be fixed, but not in the places we care about.  That makes this approach a no-go.

I could do the same thing with adapters.  Alternatively we could just throw from API that used these interfaces and not try to implement it.